woocommerce/woocommerce-gutenberg-products-block
# Products block: Using nested selector to add margin between Product elements

Our goal is to create consistent spacing between product elements within the Products block to provide an improved default user interface when adding the products block. Furthermore, we aim to maintain the Gutenberg styling hierarchy:

*Editor Styles > Global Styles > Child Theme > Parent Theme > Core*

In this pull request, I have added the `products-block-post-template` class to the `core/post-template` block, which includes all product elements. Following this, I have incorporated the subsequent CSS code into [assets/js/blocks/product-query/style.scss](https://github.com/woocommerce/woocommerce-blocks/pull/8993/files#diff-7e129284e052f95ab89530148a43d8ba7a448b15887ec2e10ac6e1b1b6482ee4):

```
:where(.products-block-post-template .wp-block-post) > * {
	margin-bottom: 0.75rem;
	margin-top: 0;
}
```

This adds default margin-top and bottom for children of `core/post-template`.

You can find more info in [comment below](https://github.com/woocommerce/woocommerce-blocks/pull/8993#issuecomment-1508048283).

Fixes #7954 




### Testing

1. Create a new post & Add Products block to it.
2. Add all product elements as inner blocks
<img width="274" alt="image" src="https://user-images.githubusercontent.com/16707866/233044525-50b864de-0236-4d11-8bdc-9e89c539b611.png">

3. Save & publish
4. Verify that on Editor & Frontend, There is equal space between the product elements.
5. Now using following steps, verify that user can still change margin-top & bottom of products elements using Global styles:
    1. Open the Global Styles sidebar by clicking the "Styles" button at the top right corner of the editor.
https://user-images.githubusercontent.com/16707866/228781997-24a9a29f-4be7-46d4-b1b3-41520692d86d.mov
    2. In the Global Styles sidebar, locate any product element, for example, "Product Price" block.
    3. Change margin-top & bottom and save the changes.
    4. Check the Product Price block in both the editor and on the front-end to ensure that the changes made through the Global styles UI are accurately applied:
        - In the editor, inspect Product Price block & ensure it matches the value set in the Global styles UI.
        - Save the post & publish it. On frontend, inspect Product Price block & ensure it matches the value set in the Global styles UI.


### WooCommerce Visibility




### Changelog

> Products block: Using nested selectors to add consistent spacing between Product elements
